src/main/java/com/akademia/educators_back/service/ProblemService.java

```diff
@@ -0,0 +1,35 @@
+package com.akademia.educators_back.service;
+
+import com.akademia.educators_back.to.ProblemTo;
+import com.akademia.educators_back.entity.ProblemEntity;
+import com.akademia.educators_back.exception.ProblemDoesNotExistException;
+import com.akademia.educators_back.mapper.ProblemMapper;
+import com.akademia.educators_back.repository.ProblemRepository;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ProblemService {
+
+    private ProblemRepository problemRepository;
+    private ProblemMapper problemMapper;
+    private ProblemEntity problemEntity;
+
+    public void addProblemToDB(ProblemTo problemTo) {
+        problemEntity = problemMapper.toProblemEntity(problemTo);
+        problemRepository.save(problemEntity);
+    }
+
+    public void deleteProblemFromDB(ProblemTo problemTo) {
+        problemEntity = problemMapper.toProblemEntity(problemTo);
+        problemRepository.delete(problemEntity);
+    }
+
+    public void updateProblem(ProblemTo problemTo) {
+        problemEntity = problemMapper.toProblemEntity(problemTo);
+        problemEntity = problemRepository.findById(problemTo.getId()).orElseThrow(()->new ProblemDoesNotExistException("Problem does not exist"));
+        problemEntity.setQuestion(problemTo.getQuestion());
```

Comment:
Co by się stało gdybyśmy chcieli zrobić update tylko jednego pola? Obiekt TO może mieć pola nullowe. W następnych zadaniach dodamy walidatory, póki co zostawimy.